src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTask.java

```diff
@@ -218,6 +193,29 @@ private void processSinkRecords(final MongoSinkTopicConfig config, final List<Si
     }
   }
 
+  private void resetRemainingRetriesForTopic(final MongoSinkTopicConfig topicConfig) {
+    getRemainingRetriesForTopic(topicConfig.getTopic())
+        .set(topicConfig.getInt(MAX_NUM_RETRIES_CONFIG));
+  }
+
+  private void checkRateLimit(final MongoSinkTopicConfig config) {
+    RateLimitSettings rls = config.getRateLimitSettings();
+
+    if (rls.isTriggered()) {
+      LOGGER.debug(
+          "Rate limit settings triggering {}ms defer timeout after processing {}"
+              + " further batches for topic {}",
+          rls.getTimeoutMs(),
+          rls.getEveryN(),
+          config.getTopic());
+      try {
+        Thread.sleep(rls.getTimeoutMs());
+      } catch (InterruptedException e) {
+        LOGGER.warn(e.getMessage());
```

Comment:
Please don't hate me for this, but I continue saying that swallowing interrupt signals is not a correct way to handle them.

In this specific case it appears that re-throwing the `InterruptedException` is the best way to handle it. Then in the method `bulkWriteBatch` we could handle it by
- re-throwing it wrapped into an unchecked exception and re-setting the interrupt flag;
- optionally additionally logging the fact that writing documents was interrupted.

Reply:
No problems, this is code / rate limiting was implemented before we took on the library but I do need to understand more about this for my own knowledge outside of this PR. 

Can you double check e13f2d2 does the right thing?

Reply:
Thanks! I started a discussion on the new code: https://github.com/mongodb/mongo-kafka/pull/50#discussion_r569649044.